### PR TITLE
Audit Log REST API(Beta) Implementation

### PR DESCRIPTION
**Pull request by sridharavinash**
This PR implements the new AuditLog REST API Beta which was GA-ed recently. 

You can read more about it here: 
https://github.blog/changelog/2021-02-03-audit-log-git-events-and-rest-api-are-now-in-public-beta/

Most notably it exposes 2 new endpoints for Enterprises Accounts only 
* `orgs/<org>/audit-log`
* `enterprises/<enterprise>/audit-log` 

In this PR
* Extend `OrganizationsService` to implement `GetAuditLog` call
* Extend `EnterpriseService` to implement `GetAuditLog` call
* Tests for both services

I chose to put these into 2 separate files because they define 2 separate endpoints (I can be persuaded to have this live in one file if it makes it easier). 
API reference docs : https://docs.github.com/en/rest/reference/orgs#get-the-audit-log-for-an-organization

### CODE REVIEW

**Comment by gmlewis** [comment]
Thank you, @sridharavinash - it looks like you might need to run `go generate ./...` as described in our CONTRIBUTING.md guide.

Hey, by the way... do you know what happened to this blog site: https://developer.github.com/changes/ ?  I haven't seen any changes on it since October 1, 2020, and your PR is the first I've seen of a different blog site URL.

**Comment by sridharavinash** [comment]
>Thank you, @sridharavinash - it looks like you might need to run go generate ./... as described in our CONTRIBUTING.md guide.

@gmlewis 👋🏾 . I think this was `fmt` failure. Hopefully the last commit fixes that. 

>Hey, by the way... do you know what happened to this blog site: https://developer.github.com/changes/ ? I haven't seen any changes on it since October 1, 2020, and your PR is the first I've seen of a different blog site URL.

Good question. I'll find out more if that's being deprecated. Maybe in favor of OpenAPI descriptions 
 - https://github.com/github/rest-api-description 🤔 .

File: github/enterprise_audit_log.go
Comment (gmlewis) [inline github/enterprise_audit_log.go:15]: I think this URL should be different for the Enterprise version.

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:24]: I'm extremely confused at the moment.
When I look at the examples listed here:
https://docs.github.com/en/rest/reference/orgs#get-the-audit-log-for-an-organization

I see three audit entry examples, with just a tiny handful of fields.

Where is this huge list of fields coming from?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @sridharavinash !
Please see comments below.

File: github/enterprise_audit_log.go
Comment (sridharavinash) [reply github/enterprise_audit_log.go:15]: Fixed, I had that as a placeholder till I found the real link.

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:24]: We have an internal (not yet public) representation of the these exposed API fields for the 600+ events that the audit-log supports.  Some of these are defined here: https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/reviewing-the-audit-log-for-your-organization#audit-log-actions (not all these actions will be available in an org/enterprise setting).

Comment (sridharavinash) [reply github/orgs_audit_log.go:24]: Thinking out aloud should we rather have these be more tightly coupled to the Action and then do a lookup based on `action` or maybe embed other Actions event fields?

so instead of 
```
type AuditEntry struct {
     <smattering of fields>
} 
```
do something more akin to 
```
type AuditEntry struct{
   Action *string `json:"action,omitempty"`
    DocumentID *string `json:"_document_id,omitempty"`
    Timestamp *int64 `json:"*timestamp,omitempty"`

    GitEventAuditEntry
    TeamEventAuditEntry
}

type GitEventAuditEntry struct {
   TransportProtcol *string `json:"transport_protocol,omitempty"`
    ....
}
```

File: github/org_audit_log_test.go
Comment (sridharavinash) [reply github/org_audit_log_test.go]: fixed in https://github.com/google/go-github/pull/1820/commits/1d91016ead0af1f0a964de0841fcaef68c64ffe3

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:14]: @gmlewis Would you recommend that I move `After` and `Before` to `ListCursorOptions` 
IOW
```
type GetAuditLogOptions struct {
	Phrase  *string `json:"phrase,omitempty"`   // A search phrase. (Optional.)
	Include *string `json:"include,omitempty"`  // Event type includes. Can be one of "web", "git", "all". Default: "web". (Optional.)
	Order   *string `json:"order,omitempty"`    // The order of audit log events. Can be one of "asc" or "desc". Default: "desc". (Optional.)

         // Pagination Options
	ListCursorOptions
}
```

and update 
```
// ListCursorOptions specifies the optional parameters to various List methods that
// support cursor pagination.
type ListCursorOptions struct {
	// For paginated result sets, page of results to retrieve.
	Page string `url:"page,omitempty"`

	// For paginated result sets, the number of results to include per page.
	PerPage int `url:"per_page,omitempty"`
        
        // A cursor, as given in the Link header. If specified, the query only searches for events after this cursor. (Optional.)
	After   string `url:"after,omitempty"`    

        // A cursor, as given in the Link header. If specified, the query only searches for events before this cursor. (Optional.)
	Before  string `url:"before,omitempty"`   

}
```

Comment (gmlewis) [reply github/orgs_audit_log.go:14]: @sridharavinash - in the past, I think we decided that the `After` and `Before` URLs were not useful to users of this client library, so we have explicitly not added them.

So unless you have any objections, I think we can just drop them.

File: github/orgs_audit_log.go
Comment (gmlewis) [reply github/orgs_audit_log.go:24]: Hmm... great question.  I'm fine with making one huge struct like we do for some of the other responses, but I also see the value of your idea...

I suppose we might want to consider how challenging one vs the other might be for maintenance of this repo as things change on the GitHub side.

How about you implement what makes sense to you, then we can see what it looks like and others can comment if they have an opinion as well.

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:14]: Unfortunately for the AuditLog API we don't have a `page` param, we exclusively use `before` / `after` link `rel` headers to navigate pages. Using the API call with `PerPage` will probably be limited to just viewing one page without any forward/backwards navigation of results. 

I can drop it if that's preferred, it can always be added if someone needs it.

Comment (gmlewis) [reply github/orgs_audit_log.go:14]: Ah!  OK, @sridharavinash - then in that case if it is going to be useful, go ahead and add it.

Now the question is whether or not it belongs in `ListCursorOptions`, and I'm guessing the answer is "yes", so go ahead and include it.

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:23]: I went down this path of separating the fields into logical structs and decided against it only because I think discoverability of the fields might suffer and not all fields map 1:1 to an audit-action event. Instead I sorted the field list within here so that it's easily discoverable and gives the user a comprehensive look of what can be expected back.

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:14]: added in https://github.com/google/go-github/pull/1820/commits/1814994b6d8b1ed35ae0ae09b6a5e34418c8f2f1

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:74]: To minimize confusion and maximize consistency, could we please change this from `TransportName` to `TransportProtocolName`?

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:18]: Could you please add a blank line above `ListCursorOptions`?

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:42]: Just FYI - I'm going to just trust you on the types since this is not officially documented yet.
As usual, we can always update these when they change.

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:65]:
```suggestion
	RunnerLabels         []string  `json:"runner_labels,omitempty"`
```

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:67]:
```suggestion
	SecretsPassed        []string  `json:"secrets_passed,omitempty"`
```

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:75]: `TransportProtocol` is really an `*int64` ?  That seems really odd to me.

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:72]: A `Team` doesn't point to a `*Team`?  It's really just a string?

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:77]: A `User` doesn't point to a `*User` and is just a string?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @sridharavinash !
I have a few questions, and a few minor tweaks, please.

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:75]: good catch! I think this can be just `*int`

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:72]: that's correct, it's just a string of the team name. Since the Audit Log is mostly immutable logs, we don't store full object details (in the event that the team / repo / user account has been deleted for example).

File: github/orgs_audit_log.go
Comment (sridharavinash) [reply github/orgs_audit_log.go:77]: Same as https://github.com/google/go-github/pull/1820#discussion_r593277513

**Comment by sridharavinash** [comment]
I'm going to update this PR with a little more detail and add documentation of the returned types where possible.

**Comment by sridharavinash** [comment]
I made a few changes and addressed some of the feedback. Let me know how this is looking @gmlewis . 🙇🏾‍♂️

**Review by gmlewis — APPROVED** [review]
Thank you, @sridharavinash !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp .
Merging.

**Pull request merged**
